rllib/algorithms/iql/iql.py

```diff
@@ -0,0 +1,202 @@
+        value_lr: Optional[LearningRateOrSchedule] = NotProvided,
+        target_network_update_freq: Optional[int] = NotProvided,
+        tau: Optional[float] = NotProvided,
+        **kwargs,
+    ) -> "IQLConfig":
+        super().training(**kwargs)
+
+        if twin_q is not NotProvided:
+            self.twin_q = twin_q
+        if expectile is not NotProvided:
+            self.expectile = expectile
+        if actor_lr is not NotProvided:
+            self.actor_lr = actor_lr
+        if critic_lr is not NotProvided:
+            self.critic_lr = critic_lr
+        if value_lr is not NotProvided:
+            self.value_lr = value_lr
+        if target_network_update_freq is not NotProvided:
+            self.target_network_update_freq = target_network_update_freq
+        if tau is not NotProvided:
+            self.tau = tau
+
+        return self
+
+    @override(AlgorithmConfig)
+    def get_default_learner_class(self) -> Union[Type["Learner"], str]:
+        if self.framework_str == "torch":
+            from ray.rllib.algorithms.iql.torch.iql_torch_learner import IQLTorchLearner
+
+            return IQLTorchLearner
+        else:
+            raise ValueError(
+                f"The framework {self.framework_str} is not supported. "
+                "Use `'torch'` instead."
+            )
+
+    @override(AlgorithmConfig)
+    def get_default_rl_module_spec(self) -> RLModuleSpecType:
+        if self.framework_str == "torch":
+            from ray.rllib.algorithms.iql.torch.default_iql_torch_rl_module import (
+                DefaultIQLTorchRLModule,
+            )
+
+            return RLModuleSpec(module_class=DefaultIQLTorchRLModule)
+        else:
+            raise ValueError(
+                f"The framework {self.framework_str} is not supported. "
+                "Use `torch` instead."
+            )
+
+    @override(AlgorithmConfig)
+    def build_learner_connector(
+        self,
+        input_observation_space,
+        input_action_space,
+        device=None,
+    ):
+        pipeline = super().build_learner_connector(
+            input_observation_space=input_observation_space,
+            input_action_space=input_action_space,
+            device=device,
+        )
+
+        # Remove unneeded connectors from the MARWIL connector pipeline.
+        pipeline.remove("AddOneTsToEpisodesAndTruncate")
+        pipeline.remove("GeneralAdvantageEstimation")
+
+        # Prepend the "add-NEXT_OBS-from-episodes-to-train-batch" connector piece (right
+        # after the corresponding "add-OBS-..." default piece).
+        pipeline.insert_after(
+            AddObservationsFromEpisodesToBatch,
+            AddNextObservationsFromEpisodesToTrainBatch(),
+        )
+
+        return pipeline
+
+    @override(MARWILConfig)
+    def validate(self) -> None:
+        # Call super's validation method.
+        super().validate()
+
+        # Ensure hyperparameters are meaningful.
+        if self.beta == 0.0:
```

Comment:
super nit: Should we check for beta > 0.0 then instead?

Reply:
Again, great catch! Yes we should!

Reply:
Great catch! Makes totally sense haha